### PR TITLE
Partial docs fixes in Newsletter module

### PR DESCRIPTION
**Pull request by ghost**
### Description
Small PHPDocs fixes in Newsletter 
strict_type fixes in Newsletter module

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @SikailoISM. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

File: app/code/Magento/Newsletter/Controller/Manage/Save.php [hunk 1]
```diff
@@ -110,7 +111,7 @@ public function execute()
                 $this->messageManager->addError(__('Something went wrong while saving your subscription.'));
             }
         }
-        $this->_redirect('customer/account/');
+        return $this->_redirect('customer/account/');
```
Comment (orlangur) [inline app/code/Magento/Newsletter/Controller/Manage/Save.php:114]: Why this is needed?

**Review by orlangur — CHANGES_REQUESTED** [review]

File: app/code/Magento/Newsletter/Controller/Manage/Save.php (hunk 1 above)
Comment (ghost) [reply app/code/Magento/Newsletter/Controller/Manage/Save.php:114]: I think this is logical due to lib/internal/Magento/Framework/App/ActionInterface.php:30

**Review by orlangur — APPROVED** [review]

File: app/code/Magento/Newsletter/Controller/Manage/Save.php [hunk 2]
```diff
@@ -65,9 +66,9 @@ public function __construct(
     /**
      * Save newsletter subscription preference action
      *
-     * @return void|null
+     * @return \Magento\Framework\App\ResponseInterface
      */
-    public function execute()
+    public function execute(): \Magento\Framework\App\ResponseInterface
```
Comment (sidolov) [inline app/code/Magento/Newsletter/Controller/Manage/Save.php]: Unfortunately, we can't change public methods signature according to our [Backward Compatible Development Guide](https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/)

File: app/code/Magento/Newsletter/Controller/Subscriber/Confirm.php [hunk 3]
```diff
@@ -15,11 +17,11 @@
 class Confirm extends \Magento\Newsletter\Controller\Subscriber implements HttpGetActionInterface
 {
     /**
-     * Subscription confirm action.
+     * Subscription confirm action
      *
      * @return \Magento\Framework\Controller\Result\Redirect
      */
-    public function execute()
+    public function execute(): \Magento\Framework\Controller\Result\Redirect
```
Comment (sidolov) [inline app/code/Magento/Newsletter/Controller/Subscriber/Confirm.php]: Unfortunately, we can't change public methods signature according to our [Backward Compatible Development Guide](https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/)

**Review by sidolov — CHANGES_REQUESTED** [review]

**Comment by orlangur** [comment]
@sidolov which sentence are you referring to? Didn't find appropriate searching by "type".

Also, these are controller actions and unlikely to be extended.

**Comment by sidolov** [comment]
Hi @orlangur , defining the strict return type is method signature changes actually, so, we can't make such changes for public methods according to our guides.

**Comment by orlangur** [comment]
@sidolov I don't see which part of the doc you're referring to. Closest one:

> Some changes are detected and documented by an automated tool. These backward incompatible changes do not need manual documentation:
> Modifying a method signature

For `execute` method I don't see why we need it as it is unlikely to be extended.

File: app/code/Magento/Newsletter/Controller/Manage/Save.php (hunk 2 above)
Comment (p-bystritsky) [reply app/code/Magento/Newsletter/Controller/Manage/Save.php]: @sidolov fixed.

File: app/code/Magento/Newsletter/Controller/Subscriber/Confirm.php (hunk 3 above)
Comment (p-bystritsky) [reply app/code/Magento/Newsletter/Controller/Subscriber/Confirm.php]: @sidolov fixed.

**Comment by p-bystritsky** [comment]
@sidolov please review.

**Review by orlangur — CHANGES_REQUESTED** [review]
@SikailoISM please squash changes into a single commit so that we have perfectly clean history 😉

**Review by sidolov — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @sidolov, thank you for the review.
[ENGCOM-4676](https://jira.corp.magento.com/browse/ENGCOM-4676) has been created to process this Pull Request

**Review by orlangur — CHANGES_REQUESTED** [review]
Please squash changes into a single commit so that we have perfectly clean history 😉

**Comment by p-bystritsky** [comment]
@orlangur updated.

**Review by orlangur — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @orlangur, thank you for the review.
[ENGCOM-4676](https://jira.corp.magento.com/browse/ENGCOM-4676) has been created to process this Pull Request

**Comment by soleksii** [comment]
:heavy_check_mark: QA Passed

**Pull request merged**